mlflow/data/huggingface_dataset_source.py

```diff
@@ -23,53 +21,63 @@ def __init__(
-        :param data_dir: The `data_dir` of the Hugging Face dataset configuration.
-        :param data_files: Paths to source data file(s) for the Hugging Face dataset configuration.
-        :param revision: Version of the dataset script to load.
-        :param task: The task to prepare the Hugging Face dataset for during training and
-                     evaluation.
-        """
-        self._path = path
-        self._config_name = config_name
-        self._data_dir = data_dir
-        self._data_files = data_files
-        self._split = split
-        self._revision = revision
-        self._task = task
+        self.path = path
+        self.config_name = config_name
+        self.data_dir = data_dir
+        self.data_files = data_files
+        self.split = split
+        self.revision = revision
 
     @staticmethod
     def _get_source_type() -> str:
         return "hugging_face"
 
     def load(self, **kwargs):
-        """
-        Loads the dataset source as a Hugging Face Dataset.
-
-        :param kwargs: Additional keyword arguments used for loading the dataset with
-                       the Hugging Face ``datasets.load_dataset()`` method. The following keyword
-                       arguments are used automatically from the dataset source but may be
-                       overridden by values passed in ``**kwargs``: ``path``, ``name``,
-                       ``data_dir``, ``data_files``, ``split``, ``revision``, ``task``.
-        :return: An instance of ``datasets.Dataset``.
+        """Load the Hugging Face dataset based on `HuggingFaceDatasetSource`.
+
+        Args:
+            kwargs: Additional keyword arguments used for loading the dataset with the Hugging Face
+                `datasets.load_dataset()` method.
+
+        Returns:
+            An instance of `datasets.Dataset`.
         """
         import datasets
 
         load_kwargs = {
-            "path": self._path,
-            "name": self._config_name,
-            "data_dir": self._data_dir,
-            "data_files": self._data_files,
-            "split": self._split,
-            "revision": self._revision,
-            "task": self._task,
+            "path": self.path,
+            "name": self.config_name,
+            "data_dir": self.data_dir,
+            "data_files": self.data_files,
+            "split": self.split,
+            "revision": self.revision,
         }
-        load_kwargs.update(kwargs)
-
+        for k, v in kwargs.items():
+            if load_kwargs.get(k, None):
+                raise KeyError(
+                    f"Found duplicated arguments {k} in `HuggingFaceDatasetSource` and `kwargs`: "
+                    f"`HuggingFaceDatasetSource` has `{k}={v}`, and `kwargs` has `{k}={kwargs[k]}`"
+                    f"Please remove {k} from `kwargs`."
+                )
+            # Copy `kwargs` to `load_kwargs`.
+            load_kwargs[k] = v
```

Comment:
Can we use set operations here to be more consistent with other similar operations within MLflow?

```suggestion
        intersecting_keys = set(load_kwargs.keys()) & set(kwargs.keys())
        if intersecting_keys:
            raise KeyError(
                f"Found duplicated arguments in `HuggingFaceDatasetSource` and `kwargs`: {intersecting_keys}. "
                f"Please remove them from `kwargs`."
            )
            load_kwargs.update(kwargs)
```

Validation:
``` python
class HuggingFaceDatasetSource:
    def __init__(self):
        self.path = "path_value"
        self.config_name = "config_name_value"
        self.data_dir = "data_dir_value"
        self.data_files = "data_files_value"
        self.split = "split_value"
        self.revision = "revision_value"
        
    def load(self, **kwargs):
        """Load the Hugging Face dataset based on `HuggingFaceDatasetSource`.
        Args:
            kwargs: Additional keyword arguments used for loading the dataset with the Hugging Face
                `datasets.load_dataset()` method.
        Returns:
            An instance of `datasets.Dataset`.
        """
        import datasets

        load_kwargs = {
            "path": self.path,
            "name": self.config_name,
            "data_dir": self.data_dir,
            "data_files": self.data_files,
            "split": self.split,
            "revision": self.revision,
        }

        intersecting_keys = set(load_kwargs.keys()) & set(kwargs.keys())
        if intersecting_keys:
            raise KeyError(
                f"Found duplicated arguments in `HuggingFaceDatasetSource` and `kwargs`: {intersecting_keys}. "
                f"Please remove them from `kwargs`."
            )

        load_kwargs.update(kwargs)

def validate_load_method():
    dataset_source = HuggingFaceDatasetSource()
    try:
        dataset_source.load(new_key_1="value_1", new_key_2="value_2")
        print("Test with no overlapping kwargs: PASSED, as expected")
    except KeyError as e:
        print(f"ALERT! Test with no overlapping kwargs: FAILED. Error: {e}")
    
    try:
        dataset_source.load(path="new_path_value", new_key_3="value_3")
        print("ALERT! Test with overlapping kwargs: PASSED")
    except KeyError as e:
        print(f"Test with overlapping kwargs: FAILED, as expected. Error: {e}")

validate_load_method()
```

Reply:
Sure thing! thanks for the suggestion!